packages/babel-generator/src/whitespace.js

```diff
@@ -42,19 +17,11 @@ export default class Whitespace {
     let endToken;
     let tokens = this.tokens;
 
-    for (let j = 0; j < tokens.length; j++) {
-      // optimize for forward traversal by shifting for loop index
-      let i = getLookupIndex(j, this._lastFoundIndex, this.tokens.length);
-      let token = tokens[i];
-
-      // this is the token this node starts with
-      if (node.start === token.start) {
-        startToken = tokens[i - 1];
-        endToken = token;
-
-        this._lastFoundIndex = i;
-        break;
-      }
+    let index = this._findToken(token => token.start - node.start, 0, tokens.length);
+    if (typeof index === "number") {
+      while (index && node.start === tokens[index - 1].start) --index;
```

Comment:
ah right because startToken is optional in `getNewlinesBetween`? Can we change it to `index >= 0`? If `_findToken` can return `-1` then this condition will be truthy for `-1` which seems wrong.